compiler/rustc_middle/src/ty/mod.rs

```diff
@@ -424,11 +425,15 @@ crate struct TyS<'tcx> {
     /// De Bruijn indices within the type are contained within `0..D`
     /// (exclusive).
     outer_exclusive_binder: ty::DebruijnIndex,
+
+    /// The stable hash of the type. This way hashing of types will not have to work
+    /// on the address of the type anymore, but can instead just read this field
+    stable_hash: Fingerprint,
 }
 
 // `TyS` is used a lot. Make sure it doesn't unintentionally get bigger.
 #[cfg(all(target_arch = "x86_64", target_pointer_width = "64"))]
-static_assert_size!(TyS<'_>, 40);
+static_assert_size!(TyS<'_>, 56);
```

Comment:
This makes me wonder if you should do a perf run with the size increase, but not actually changing the interning or `StableHash` behavior - the regressions could come from just this.

Reply:
I checked all the regressions with cachegrind, and they are very obviously additional hashing that is happening and not anything unrelated.